clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -813,21 +816,18 @@ public void commit(final Map<TopicPartition, Long> offsets, CommitType commitTyp
      * rebalance and also on startup. As such, if you need to store offsets in anything other than Kafka, this API
      * should not be used.
      * <p>
-     * Asynchronous commits (i.e. {@link CommitType#ASYNC} will not block. Any errors encountered during an asynchronous
-     * commit are either passed to the callback (if provided) or silently discarded. Synchronous commits (i.e.
-     * {@link CommitType#SYNC}) block until either the commit succeeds or an unrecoverable error is encountered. In
-     * this case, the error is either passed to the callback (if provided) or thrown to the caller.
+     * This is an asynchronous call and will not block. Any errors encountered are either passed to the callback
+     * (if provided) or discarded.
      *
      * @param offsets The list of offsets per partition that should be committed to Kafka.
-     * @param commitType Control whether the commit is blocking
      * @param callback Callback to invoke when the commit completes
      */
     @Override
-    public void commit(final Map<TopicPartition, Long> offsets, CommitType commitType, ConsumerCommitCallback callback) {
+    public void commit(final Map<TopicPartition, Long> offsets, OffsetCommitCallback callback) {
```

Comment:
I realize it's clearly documented in the javadoc, but I am a little concerned about the naming. If you just glance at the options for committing and think "I don't need a callback, I'll just use the version that doesn't include it", you can end up with synchronous commits when what you really wanted was async without a callback, especially if you've only read the javadoc for the version that has the callback.

Although I think it's uglier, I'd suggest naming the one we think will be more commonly used as just `commit` and the other one with sync/async in the name. Then just scanning the APIs makes it clear you need to be careful which one you use. We can do without the renaming, but I suspect we'll end up with lots of mailing list questions on the topic if we leave it as is.